internal/guest/storage/scsi/scsi.go

```diff
@@ -175,27 +217,48 @@ func Mount(
 			}
 		}
 		source = encryptedSource
-		mountType = "xfs"
 	}
 
-	for {
-		if err := unixMount(source, target, mountType, flags, data); err != nil {
-			// The `source` found by GetDevicePath can take some time
-			// before its actually available under `/dev/sd*`. Retry while we
-			// wait for `source` to show up.
-			if errors.Is(err, unix.ENOENT) || errors.Is(err, unix.ENXIO) {
-				select {
-				case <-ctx.Done():
-					log.G(ctx).Warnf("mount system call failed with %s, context timed out while retrying", err)
-					return err
-				default:
-					time.Sleep(10 * time.Millisecond)
-					continue
+	// Get the filesystem that is already on the device (if any) and use that
+	// as the mountType unless `Filesystem` was given.
+	deviceFS, err := _getDeviceFsType(source)
```

Comment:
It would be good to log what filesystem we detect.

Reply:
Added